app/src/main/java/com/github/fribourgsdp/radio/StringComparisons.kt

```diff
@@ -0,0 +1,98 @@
+package com.github.fribourgsdp.radio
+
+import java.text.Normalizer
+
+const val MAX_ALLOWED_ERRORS = 2
+
+class StringComparisons {
+    companion object{
+        private const val chars = "abcdefghijklmnopqrstuvwxyz1234567890"
+
+        /**
+        * Function that returns a number of points depending on the number of mistakes in the comparison of the two provided strings.
+        */
+        fun compareAndGetPoints(actual : String, expected : String, maxPts : Int = 100) : Int {
+            val oneErrorPenalty = 0.9
+            val twoErrorsPenalty = 0.8
+            return when(compare(actual, expected)){
+                0 -> maxPts
+                1 -> (oneErrorPenalty*maxPts).toInt()
+                2 -> (twoErrorsPenalty*maxPts).toInt()
+                else -> 0
+            }
+        }
+
+        /**
+        * Function that compares two strings and returns the number of differences between them.
+        * Returns -1 if the number of differences is bigger than 2, to avoid taking too much time.
+        */
+        fun compare(actual : String, expected : String) : Int {
+            val s1 = actual.replace(Regex("[!-/]|[:-@]|[\\[-`]|[{-~]| +"), "").lowercase().unaccent()
+            val s2 = expected.replace(Regex("[!-/]|[:-@]|[\\[-`]|[{-~]| +"), "").lowercase().unaccent()
+            for (tolerance in 0..MAX_ALLOWED_ERRORS) {
+                if( equal(s1, s2, tolerance)){
+                    return tolerance
+                }
+            }
+            return -1
+        }
+
+        /**
+        * Function used to remove diacritics from a string.
+        */
+        private fun CharSequence.unaccent() : String{
+            val temp = Normalizer.normalize(this, Normalizer.Form.NFD)
+            return "\\p{InCombiningDiacriticalMarks}+".toRegex().replace(temp, "")
+        }
+        /**
+        * Recursive function that determines if two formatted strings are equal at a given level of tolerance.
+        */
+        private fun equal(s1 : String, s2 : String, tolerance: Int) : Boolean {
+            var result : Boolean = false
+            if (tolerance >= 0){
+                if (s1 == s2){
+                    result = result || true
+                }
```

Comment:
Why don't you go back to true directly here?

Reply:
To reduce the number of return points of the function. Indeed, codeclimate was not agreeing with this way of coding in a previous commit.

Reply:
But to avoid execution of the rest of the function, I can put a condition in the second if, you're right.

Reply:
So why not just do `result = true` ?